src/main/scala/cromwell/engine/backend/local/LocalBackend.scala

```diff
@@ -36,6 +37,23 @@ object LocalBackend {
     }
   }
 
+  def findTempFile(root: Path, prefix: String) = {
+    val regex = s"$prefix.*\\.tmp$$".r.unanchored
+    val filesWithPrefix = Try(Files.newDirectoryStream(root)).map(
+      stream => stream.iterator().toIterator.toSeq.collect {
```

Comment:
If you used `.toList` instead of `.toSeq` (which is just making a `List` anyways) you can use real matching in the following match instead of the size checks.